projects/shared/src/lib/components/floor/floor.component.scss

```diff
@@ -0,0 +1,136 @@
+@import "../../styles-lib/colors.scss";
+
+.floor-number {
+  display: block;
+  margin-bottom: 10px;
+  text-align: center;
+  font-size: 5rem;
+}
+
+.back-button {
+  display: block;
+  min-height: 40px;
+  border: 2px solid $SecondColor;
+  background-color: $MainColor;
+  color:#fff;
+  border-radius: 5px;
+  font-size: 1.4rem;
+  cursor: pointer;
+  margin: 20px auto;
+}
+
+.floor-area {
+  position: relative;
+  max-width: 960px;
+  min-width: 450px;
+  margin: auto;
+  background-color: #CCC;
+
+  &__bg-img {
+    width: 100%;
+  }
+
+  &__floor {
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+  }
+}
+
+.floor {
+  display: flex;
+  flex-direction: column;
+
+  &__rooms {
+    height: 40%;
+  }
+
+  &__hall {
+    height: 20%;
+    box-sizing: border-box;
+    border-left: 4px solid black;
+    border-right: 4px solid black;
+  }
+}
+
+.rooms {
+  display: flex;
+  border: 2px solid black;
+
+  &__room {
+    box-sizing: border-box;
+    height: 100%;
+    width: 100%;
+    border: 2px solid black;
+  }
+}
+
+.room {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  padding: 5px;
+  position: relative;
+
+  &__number {
+    position: absolute;
+    top: 5%;
```

Comment:
I beleive this can be achieved using flex. and this comment relates to most of the height\width\top-left declarations in this file